### DISCUSSION TITLE
PRO_Unable to create ssm put parameter with multiple tag values getting error- The security token included in the request is invalid.

### DISCUSSION THREAD

**Discussion by ashameem123** [aws/aws-cli#6841] [Q&A] [answered]
Unable to create ssm put parameter with multiple tag values, getting error - "PutParameter operation: The security token included in the request is invalid."

![image](https://user-images.githubusercontent.com/54183749/160449763-494c76de-7a83-48dd-8e0c-a94a6f7c4389.png)

![image](https://user-images.githubusercontent.com/54183749/160450793-273d0818-2250-4070-a01d-40a632ef576e.png)

**Comment by kdaily** [comment, marked as answer]
Hi @ashameem123, 

Sorry to hear you're having trouble. This error is common with issues with the credentials you are supplying. Are you sure the credentials you're supplying are correct? You can confirm where you are getting credentials from by using `aws configure list`.

Can you successfully run a different command, like `aws sts get-caller-identity`?

If you are still having an issue, we would need to see debug logs, which you can get by adding `--debug` to your command. Please copy/paste the contents as a reply to this issue, and be sure to remove any sensitive information, like account numbers, from the output. Thanks!

**Comment by ashameem123** [comment]
Hi,

This is what I'm getting in response to the command you sent above:

[image: image.png]

**Comment by ashameem123** [comment]
@kdaily - I was able to successfully create ssm parameters with multiple tags but keep getting exception:(Please have a look into this ticket)  - https://github.com/localstack/localstack/issues/5768

Exception: null (Service: Ssm, Status Code: 404, Request ID: null, Extended Request ID: null): corp.aws.exception.NonRetryableException\ncorp.aws.exception.corpNonRetryableException: Error retrieving parmeter comp-template-attributes from parameter store. 


We can close #6823 as  I'm able to create parameters first by creating parameter with value and then adding tags to it. (2 separate steps)

**Comment by ashameem123** [comment]
@kdaily - Just for better understanding, What do you mean by "invalid credentials" here?  I have a default credentials set in my .aws credentials file (aws_access_key_id = foo, aws_secret_access_key = bar)Shouldn't that be enough for running ssm in localstack?

**Comment by kdaily** [comment]
Hi @ashameem123,

Thanks for letting me know. We're not able to provide support for third party implementations. I'm not familiar enough with Localstack to know how credentials work there. You'll have to follow up with the maintainers there for more assistance. Thanks!
